test/unit/L1/l1LPTGateway.test.ts

```diff
@@ -632,6 +666,126 @@ describe('L1 LPT Gateway', function() {
                 withdrawAmount,
             );
       });
+
+      describe('escrow does not have sufficient funds', () => {
+        it('mints excess tokens', async () => {
+          const excessAmount = 200;
+
+          const finalizeWithdrawalTx = await l1Gateway
+              .connect(mockBridgeEOA)
+              .finalizeInboundTransfer(
+                  token.address,
+                  sender.address,
+                  sender.address,
+                  escrowBalance + excessAmount,
+                  defaultWithdrawData,
+              );
+
+          expect(token.transferFrom).to.be.calledWith(
+              escrow.address,
+              sender.address,
+              escrowBalance,
+          );
+
+          expect(minterMock.bridgeMint).to.be.calledWith(
+              sender.address,
+              excessAmount,
+          );
+
+          await expect(finalizeWithdrawalTx)
+              .to.emit(l1Gateway, 'WithdrawalFinalized')
+              .withArgs(
+                  token.address,
+                  sender.address,
+                  sender.address,
+                  expectedTransferId,
+                  escrowBalance + excessAmount,
+              );
+        });
+
+        it('mints excess tokens to 3rd Party', async () => {
+          const excessAmount = 200;
+
+          const finalizeWithdrawalTx = await l1Gateway
+              .connect(mockBridgeEOA)
+              .finalizeInboundTransfer(
+                  token.address,
+                  sender.address,
+                  receiver.address,
+                  escrowBalance + excessAmount,
+                  defaultWithdrawData,
+              );
+
+          expect(token.transferFrom).to.be.calledWith(
+              escrow.address,
+              receiver.address,
+              escrowBalance,
+          );
+
+          expect(minterMock.bridgeMint).to.be.calledWith(
+              receiver.address,
+              excessAmount,
+          );
+
+          await expect(finalizeWithdrawalTx)
+              .to.emit(l1Gateway, 'WithdrawalFinalized')
+              .withArgs(
+                  token.address,
+                  sender.address,
+                  receiver.address,
+                  expectedTransferId,
+                  escrowBalance + excessAmount,
+              );
+        });
+
+        describe('escrow has 0 balance', () => {
+          beforeEach(async function() {
+            // burn escrow balance
+            const balance = await token.balanceOf(escrow.address);
```

Comment:
Since `token` is a `MockContract` that has all the features of fakes from smock could we simplify the logic in this hook?

```javascript
token.balanceOf.whenCalledWith(escrow.address).returns(0)
```